client/wait_job.go

```diff
@@ -0,0 +1,229 @@
+)
+
+var WaitJobCmd = &cli.Subcommand{
+	Use:   "wait-job JOB",
+	Short: "block and wait until the specified job is done or has failed",
+	Run: func(ctx context.Context, subcommand *cli.Subcommand, args []string) error {
+		return runWaitJobCmd(subcommand.Config(), args)
+	},
+}
+
+// adapted from status.go
+func runWaitJobCmd(config *config.Config, args []string) error {
+	if len(args) != 1 {
+		return errors.Errorf("Expected 1 argument: JOB")
+	}
+
+	// TODO: poll status until done or error
+	c, err := client.New("unix", config.Global.Control.SockPath)
+	if err != nil {
+		return errors.Wrapf(err, "connect to daemon socket at %q", config.Global.Control.SockPath)
+	}
+
+	jobName := args[0]
+
+	updateAndEvaluate := func() (bool, error) {
+		s, err := c.Status()
+		if err != nil {
+			return true, err
+		}
+
+		if err != nil {
+			return true, fmt.Errorf("can't get status from daemon: %w", err)
+		}
+
+		job, ok := s.Jobs[jobName]
+		if !ok {
+			return true, fmt.Errorf("job not in status: %v", jobName)
+		}
+
+		return evaluateJobStatus(job)
+	}
+	updateAndEvaluate()
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+ticker:
+	for range ticker.C {
+		done, err := updateAndEvaluate()
+		if err != nil {
+			return err
+		}
+		if done {
+			break ticker
+		}
+	}
+
+	return nil
+}
+
+func evaluateJobStatus(j *job.Status) (bool, error) {
+
+	if j.Type == job.TypePush || j.Type == job.TypePull {
+		activeStatus, ok := j.JobSpecific.(*job.ActiveSideStatus)
+		if !ok || activeStatus == nil {
+			// TODO: ignore?
+			return true, nil //error.New("ActiveSideStatus is null")
+		}
+
+		//t.renderReplicationReport(activeStatus.Replication, t.getReplicationProgressHistory(k))
+		replicationDone, err := evaluateReplicationStatus(activeStatus.Replication)
+		if err != nil {
+			return true, err
+		}
+
+		senderPruningDone, err := evaluatePruningStatus(activeStatus.PruningSender)
+		if err != nil {
+			return true, fmt.Errorf("sender %w", err)
+		}
+		receiverPruningDone, err := evaluatePruningStatus(activeStatus.PruningReceiver)
+		if err != nil {
+			return true, fmt.Errorf("receiver %w", err)
+		}
+
+		done := replicationDone && senderPruningDone && receiverPruningDone
+
+		if j.Type == job.TypePush {
+			snapperDone, err := evaluateSnapperStatus(activeStatus.Snapshotting)
+			if err != nil {
+				return true, err
+			}
+
+			done = done && snapperDone
+		}
+
+		return done, nil
+	} else if j.Type == job.TypeSnap {
+		snapStatus, ok := j.JobSpecific.(*job.SnapJobStatus)
+		if !ok || snapStatus == nil {
+			// TODO: ignore?
+			return true, nil //error.Error("SnapJobStatus is null")
+		}
+
+		pruningDone, err := evaluatePruningStatus(snapStatus.Pruning)
+		if err != nil {
+			return true, err
+		}
+
+		snapperDone, err := evaluateSnapperStatus(snapStatus.Snapshotting)
+		if err != nil {
+			return true, err
+		}
+
+		return pruningDone && snapperDone, nil
+	} else if j.Type == job.TypeSource /*|| j.Type == job.TypeSink*/ {
```

Comment:
I'm not sure what to do here. The status code doesn't even mention `job.TypeSink`.